File: packages/core/src/config/meta.ts
// チャンネルメタ情報とブランディング設定（merged の `channel` + `youtube_channel` を合成）。

import { z } from "zod";

import { snakeToCamel } from "../../internal/case.ts";

/** `channel` セクション + `youtube_channel`（Branding）の合成。 */
export const ChannelMeta = z
  .object({
    channel: z
      .object({
        // YouTube チャンネル ID（`UC...`）。未設定（空文字）時は照合をスキップ (#561)。
        channel_id: z.string().default(""),
        core_message: z.string().default(""),
        cta_subscribe: z.string().default(""),
        name: z.string(),
        short: z.string(),
        tagline: z.string().default(""),
        url: z.string(),
        youtube_handle: z.string(),
      })
      .strict(),
    // `youtube_channel` セクション（YouTube チャンネル本体設定・任意）。
    // 未設定は null。`false` は明示的な「子供向けでない」申告として保持する。
    youtube_channel: z
      .object({
        country: z.string().default(""),
        default_language: z.string().default(""),
        description: z.string().default(""),
        keywords: z.array(z.string()).default([]),
        made_for_kids: z.boolean().nullable().default(null),
        unsubscribed_trailer: z.string().default(""),
      })
      .strict()
      .prefault({}),
  })
  .transform((o) => ({
    branding: snakeToCamel(o.youtube_channel),
    channelId: o.channel.channel_id,
    channelName: o.channel.name,
    channelShort: o.channel.short,
    channelUrl: o.channel.url,
    coreMessage: o.channel.core_message,
    ctaSubscribe: o.channel.cta_subscribe,
    tagline: o.channel.tagline,
    youtubeHandle: o.channel.youtube_handle,
  }));

export type ChannelMeta = z.infer<typeof ChannelMeta>;
export type Branding = ChannelMeta["branding"];

/**
 * Branding を YouTube API / yt-channel-settings が扱う dict 形式に戻す。
 * 未設定キーは省略するが、`made_for_kids=false` は明示申告のため省略しない。
 */
export const brandingAsApiDict = (b: Branding): Record<string, unknown> => {
  const out: Record<string, unknown> = {};
  if (b.description) {
    out.description = b.description;
  }
  if (b.keywords.length > 0) {
    out.keywords = [...b.keywords];
  }
  if (b.country) {
    out.country = b.country;
  }
  if (b.defaultLanguage) {
    out.default_language = b.defaultLanguage;
  }
  if (b.unsubscribedTrailer) {
    out.unsubscribed_trailer = b.unsubscribedTrailer;
  }
  if (b.madeForKids !== null) {
    out.made_for_kids = b.madeForKids;
  }
  return out;
};

File: src/youtube_automation/configuration/meta.py
"""チャンネルメタ情報とブランディング設定の責務別 dataclass."""

from __future__ import annotations

from dataclasses import dataclass, field


@dataclass(frozen=True)
class Branding:
    """`youtube_channel` セクション。YouTube チャンネル本体設定（任意セクション）."""

    description: str = ""
    keywords: list[str] = field(default_factory=list)
    country: str = ""
    default_language: str = ""
    unsubscribed_trailer: str = ""
    made_for_kids: bool | None = None

    @classmethod
    def from_dict(cls, data: dict | None) -> "Branding":
        if not data:
            return cls()
        return cls(
            description=data.get("description", ""),
            keywords=list(data.get("keywords", [])),
            country=data.get("country", ""),
            default_language=data.get("default_language", ""),
            unsubscribed_trailer=data.get("unsubscribed_trailer", ""),
            made_for_kids=data.get("made_for_kids"),
        )

    def as_api_dict(self) -> dict:
        """YouTube API / yt-channel-settings が扱う dict 形式に戻す（未設定キーは省略）."""
        out: dict = {}
        if self.description:
            out["description"] = self.description
        if self.keywords:
            out["keywords"] = list(self.keywords)
        if self.country:
            out["country"] = self.country
        if self.default_language:
            out["default_language"] = self.default_language
        if self.unsubscribed_trailer:
            out["unsubscribed_trailer"] = self.unsubscribed_trailer
        if self.made_for_kids is not None:
            out["made_for_kids"] = self.made_for_kids
        return out


@dataclass(frozen=True)
class ChannelMeta:
    """`channel` セクション + `youtube_channel` セクション（Branding）の合成."""

    channel_name: str
    channel_short: str
    youtube_handle: str
    channel_url: str
    core_message: str = ""
    cta_subscribe: str = ""
    tagline: str = ""
    # YouTube チャンネル ID（`UC...`）。OAuth トークン取り違え防止の照合に使う (#561)。
    # 未設定（空文字）のチャンネルでは照合をスキップする。
    channel_id: str = ""
    branding: Branding = field(default_factory=Branding)
